From: Tias Guns <tias@ulyssis.org>
Date: Thu, 2 Aug 2012 22:47:31 +0200
Subject: [PATCH 1/2] add swapoff/swapon syscalls for android

Signed-off-by: Tias Guns <tias@ulyssis.org>
---
 libbb/missing_syscalls.c |   10 ++++++++++
 1 file changed, 10 insertions(+)

Index: NHellFire/libbb/missing_syscalls.c
===================================================================
--- NHellFire.orig/libbb/missing_syscalls.c
+++ NHellFire/libbb/missing_syscalls.c
@@ -40,6 +40,16 @@ int pivot_root(const char *new_root, con
 	return syscall(__NR_pivot_root, new_root, put_old);
 }
 
+int swapoff(const char *path)
+{
+  return syscall(__NR_swapoff, path);
+}
+
+int swapon(const char *path, int swapflags)
+{
+  return syscall(__NR_swapon, path, swapflags);
+}
+
 # if __ANDROID_API__ < 21
 int tcdrain(int fd)
 {
