Index: NHellFire/networking/ifconfig.c
===================================================================
--- NHellFire.orig/networking/ifconfig.c
+++ NHellFire/networking/ifconfig.c
@@ -130,13 +130,15 @@
 #endif
 
 #if ENABLE_FEATURE_IPV6
-#ifndef __BIONIC__
+# ifndef __BIONIC__
+#  if !defined(__ANDROID__) || (defined(__ANDROID__) && __ANDROID_API__ < 21)
 struct in6_ifreq {
 	struct in6_addr ifr6_addr;
 	uint32_t ifr6_prefixlen;
 	int ifr6_ifindex;
 };
-#endif
+#  endif
+# endif
 #endif
 
 /*
Index: NHellFire/networking/interface.c
===================================================================
--- NHellFire.orig/networking/interface.c
+++ NHellFire/networking/interface.c
@@ -59,11 +59,13 @@
 /*
  * This is from linux/include/net/ipv6.h
  */
+#  if !defined(__ANDROID__) || (defined(__ANDROID__) && __ANDROID_API__ < 21)
 struct in6_ifreq {
 	struct in6_addr ifr6_addr;
 	uint32_t ifr6_prefixlen;
 	unsigned int ifr6_ifindex;
 };
+#  endif
 # endif
 #endif /* HAVE_AFINET6 */
 
Index: NHellFire/networking/route.c
===================================================================
--- NHellFire.orig/networking/route.c
+++ NHellFire/networking/route.c
@@ -43,6 +43,7 @@
 //usage:     "\n	-e	Display other/more information"
 //usage:     "\n	-A inet" IF_FEATURE_IPV6("{6}") "	Select address family"
 
+#include <sys/socket.h>
 #include <net/route.h>
 #include <net/if.h>
 #include <linux/in6.h>
Index: NHellFire/shell/glob.c
===================================================================
--- NHellFire.orig/shell/glob.c
+++ NHellFire/shell/glob.c
@@ -351,7 +351,10 @@ globtilde(const char *pattern, char *pat
 		 * the password file
 		 */
 #ifndef __GLIBC__
-		if (issetugid() != 0 ||
+		if (
+#    if __ANDROID_API__ < 21
+		    issetugid() != 0 ||
+#    endif
 		    (h = getenv("HOME")) == NULL) {
 			if (((h = getlogin()) != NULL &&
 			     (pwd = getpwnam(h)) != NULL) ||
